File: android/app/src/main/java/com/estacionamento/parking/checkout/CheckoutMath.kt
package com.estacionamento.parking.checkout

import java.time.Instant
import java.time.temporal.ChronoUnit
import kotlin.math.ceil

/** Espelha Parking.Application.Checkout.CheckoutMath. */
fun computeBillableHours(entry: Instant, exit: Instant): Int {
    val sec = ChronoUnit.SECONDS.between(entry, exit)
    if (sec <= 0) return 0
    return ceil(sec / 3600.0).toInt()
}

File: frontend-web/src/lib/checkoutMath.ts
/**
 * Espelha Parking.Application.Checkout.CheckoutMath (horas faturáveis = término do período).
 * Usa Date em UTC para comparar com ISO vindo da API.
 */
export function computeBillableHours(entry: Date, exit: Date): number {
  const sec = (exit.getTime() - entry.getTime()) / 1000
  if (sec <= 0) return 0
  return Math.ceil(sec / 3600)
}
